converter/internal/prometheusconvert/prometheusconvert.go

```diff
@@ -31,12 +32,17 @@ import (
 //	discovery.kubernetes
 //	discovery.lightsail
 //	discovery.relabel
-func Convert(in []byte) ([]byte, error) {
+func Convert(in []byte) ([]byte, common.Diagnostics) {
+	var diags common.Diagnostics
+
 	promConfig, err := promconfig.Load(string(in), false, log.NewNopLogger())
 	if err != nil {
-		return nil, err
+		diags.Add(common.SeverityLevelError, err.Error())
```

Comment:
Can you wrap the error here to make it clear that the error is from reading the file (e.g., before the conversion even starts)

Something like:

```suggestion
		diags.Add(common.SeverityLevelError, fmt.Sprintf("failed to parse Prometheus config: %s", err))
```